### ISSUE TITLE
MWT Training error

### ISSUE THREAD

**Issue by nagaraju291990** [nlp-uoregon/trankit#87]
While training MWT, getting the following error.

> UDError: The concatenation of tokens in gold file and in system file differ!
> First 20 differing characters in gold file: 'PROPNவெளியேறினார்.19' and system file: 'ஓவரில்ஏ_ஏPROPNவெளியே'
> 

How to fix this.

**Comment by harisont** [comment]
Are there any updates on this? I am encountering the same problem training on one of the official (i.e. passing all validator checks) UD treebanks.

**Comment by plural-param** [comment]
Yes the issue is with data. That is in the # text metadata when all the tokens are joined with space when comapared with that of the tokens in the 2nd column of the CONLLU format for that sentence they do not match hence this error will occur. For this we have to make sure all tokens split based on space are on the 2nd column of the CONLLU format and MWT splitting is properly done along with correct token numbers.

**Comment by nagaraju291990** [comment]
Yes the issue is with data. That is in the # text metadata when all the tokens are joined with space when comapared with that of the tokens in the 2nd column of the CONLLU format for that sentence they do not match hence this error will occur. For this we have to make sure all tokens split based on space are on the 2nd column of the CONLLU format and MWT splitting is properly done along with correct token numbers.
